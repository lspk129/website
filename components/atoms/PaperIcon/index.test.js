import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Paper from 'material-ui/Paper';

import PaperIcon from './';

describe('PaperIcon', () => {
  const muiTheme = getMuiTheme();
  const mountWithContext = node => mount(node, {
    context: { muiTheme },
    childContextTypes: { muiTheme: PropTypes.object },
  });

  it('renders correctly', () => {
    const wrapper = mountWithContext(<PaperIcon />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(Paper)).toBePresent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with circle props', () => {
    const wrapper = mountWithContext(<PaperIcon circle />);
    expect(wrapper).toHaveProp('circle', true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
