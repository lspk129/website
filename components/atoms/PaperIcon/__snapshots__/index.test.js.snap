// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperIcon renders correctly 1`] = `
<Styled(Paper)>
  <Paper
    circle={false}
    className="sc-bdVaJa fGDlZB"
    rounded={true}
    transitionEnabled={true}
    zDepth={1}
  >
    <div
      className="sc-bdVaJa fGDlZB"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    />
  </Paper>
</Styled(Paper)>
`;

exports[`PaperIcon renders with circle props 1`] = `
<Styled(Paper)
  circle={true}
>
  <Paper
    circle={true}
    className="sc-bdVaJa fGDlZB"
    rounded={true}
    transitionEnabled={true}
    zDepth={1}
  >
    <div
      className="sc-bdVaJa fGDlZB"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "50%",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    />
  </Paper>
</Styled(Paper)>
`;
